File: Springboot/src/main/java/me/rainbow/controller/QrCodeController.java
package me.rainbow.controller;

import me.rainbow.util.QrCodeUtils;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.imageio.ImageIO;
import javax.servlet.http.HttpServletResponse;
import java.awt.image.BufferedImage;

/**
 * @author guojinpeng
 * @date 17.11.9 14:56
 */
@Controller
@RequestMapping("/qrCode")
public class QrCodeController {

    @RequestMapping("/get")
    public void t2(HttpServletResponse response) {
        System.out.println("get");
        try {
            response.setHeader("Pragma", "No-cache");
            response.setHeader("Cache-Control", "no-cache");
            response.setDateHeader("Expires", 0);
            BufferedImage image = QrCodeUtils.encode("hello");
            response.setCharacterEncoding("UTF-8");
            response.setContentType("image/*");
            if (image != null) {
                ImageIO.write(image, "png", response.getOutputStream());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: AdManagementSystem/src/main/java/me/rainbow/controller/QrCodeController.kt
package me.rainbow.controller

import me.rainbow.entity.Log
import me.rainbow.service.LogService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import java.util.*

/**
 *
 * @author guojinpeng
 * @date 17.12.28 15:35
 */
@RequestMapping("/qrCode")
@Controller("qrCodeController")
open class QRCodeController {
    private val log = LoggerFactory.getLogger(QRCodeController::class.java)
    @Autowired
    lateinit open var logService: LogService

    @RequestMapping(params = ["list"])
    @ResponseBody
    fun queryAll(): HashMap<String, Any?> {
        log.debug("")
        val list = logService.getAllLogs()
        val map = HashMap<String, Any?>()
        for (log in list) {
            map["ip"] = log.ip
            map["addr"] = log.addr
        }
        return map
    }

    @RequestMapping(params = ["get"])
    @ResponseBody
    fun queryOne(id: Int): Log {
        log.debug("id为: {}", id)
        return logService.getLog(id)
    }
}